src/utils/ucc_math.h

```diff
@@ -143,11 +143,22 @@ static inline uint32_t ucc_ilog2_ceil(uint32_t n)
 /* return the lowest greater or equal power of 2 */
 static inline int ucc_lowest_greater_power2(int value)
 {
-    int p = 1;
-    while (p < value) {
-        p *= 2;
+    if (value <= 1) {
+        return 1;
     }
-    return p;
+
+    // Handle integer overflow case
+    if (value > (INT32_MAX / 2)) {
+        return INT32_MAX;  
+    }
+
+    --value;  // pow2 case
```

Comment:
I am not sure to understand the comment

Reply:
corner case if value is already power of two?

Reply:
I see, so for we need calculate value **strictly** greater than input, fo 4 it should be 8

Reply:
Do as you think is best -- personally I would leave without any comment or would write "decreasing by 1 is necessary to handle the case where value is a power of 2"

Reply:
nvm that's outdated